aws-auth-cognito/src/androidTest/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPluginEmailMFATests.kt

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.auth.cognito
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import com.amplifyframework.api.aws.AWSApiPlugin
+import com.amplifyframework.api.graphql.SimpleGraphQLRequest
+import com.amplifyframework.auth.AuthUserAttribute
+import com.amplifyframework.auth.AuthUserAttributeKey
+import com.amplifyframework.auth.MFAType
+import com.amplifyframework.auth.cognito.exceptions.service.CodeMismatchException
+import com.amplifyframework.auth.cognito.test.R
+import com.amplifyframework.auth.cognito.testutils.AbortableCountdownLatch
+import com.amplifyframework.auth.options.AuthSignUpOptions
+import com.amplifyframework.auth.result.AuthSignUpResult
+import com.amplifyframework.auth.result.step.AuthSignInStep
+import com.amplifyframework.core.configuration.AmplifyOutputs
+import com.amplifyframework.core.configuration.AmplifyOutputsData
+import com.amplifyframework.datastore.generated.model.MfaInfo
+import com.amplifyframework.testutils.Assets
+import com.amplifyframework.testutils.sync.SynchronousAuth
+import java.util.Random
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import org.junit.After
+import org.junit.BeforeClass
+import org.junit.Test
+
+class AWSCognitoAuthPluginEmailMFATests {
+
+    private val password = "${UUID.randomUUID()}BleepBloop1234!"
+    private val userName = "test${Random().nextInt()}"
+    private val email = "$userName@testdomain.com"
+
+    companion object {
+        private var authPlugin = AWSCognitoAuthPlugin()
+        private var apiPlugin = AWSApiPlugin()
+        lateinit var synchronousAuth: SynchronousAuth
+        var mfaCode = ""
+        var abortableLatch: AbortableCountdownLatch? = null
+
+        @JvmStatic
+        @BeforeClass
+        fun initializePlugin() {
```

Comment:
Can we instead start cleanly each time? We've made efforts to be able to do so. This should work as long as no code is calling an `Amplify` singleton.